solidity/contracts/BondedECDSAKeep.sol

```diff
@@ -147,361 +37,15 @@ contract BondedECDSAKeep is IBondedECDSAKeep {
-        require(!isSigningInProgress(), "Signer is busy");
-
-        /* solium-disable-next-line */
-        signingStartTimestamp = block.timestamp;
-
-        digests[_digest] = block.number;
-        digest = _digest;
-
-        emit SignatureRequested(_digest);
-    }
-
-    /// @notice Checks if keep is currently awaiting a signature for the given digest.
-    /// @dev Validates if the signing is currently in progress and compares provided
-    /// digest with the one for which the latest signature was requested.
-    /// @param _digest Digest for which to check if signature is being awaited.
-    /// @return True if the digest is currently expected to be signed, else false.
-    function isAwaitingSignature(bytes32 _digest) external view returns (bool) {
-        return isSigningInProgress() && digest == _digest;
-    }
-
-    /// @notice Submits a signature calculated for the given digest.
-    /// @dev Fails if signature has not been requested or a signature has already
-    /// been submitted.
-    /// Validates s value to ensure it's in the lower half of the secp256k1 curve's
-    /// order.
-    /// @param _r Calculated signature's R value.
-    /// @param _s Calculated signature's S value.
-    /// @param _recoveryID Calculated signature's recovery ID (one of {0, 1, 2, 3}).
-    function submitSignature(
-        bytes32 _r,
-        bytes32 _s,
-        uint8 _recoveryID
-    ) external onlyMember {
-        require(isSigningInProgress(), "Not awaiting a signature");
-        require(_recoveryID < 4, "Recovery ID must be one of {0, 1, 2, 3}");
-
-        // Validate `s` value for a malleability concern described in EIP-2.
-        // Only signatures with `s` value in the lower half of the secp256k1
-        // curve's order are considered valid.
-        require(
-            uint256(_s) <=
-                0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0,
-            "Malleable signature - s should be in the low half of secp256k1 curve's order"
-        );
-
-        // We add 27 to the recovery ID to align it with ethereum and bitcoin
-        // protocols where 27 is added to recovery ID to indicate usage of
-        // uncompressed public keys.
-        uint8 _v = 27 + _recoveryID;
-
-        // Validate signature.
-        require(
-            publicKeyToAddress(publicKey) == ecrecover(digest, _v, _r, _s),
-            "Invalid signature"
-        );
-
-        signingStartTimestamp = 0;
-
-        emit SignatureSubmitted(digest, _r, _s, _recoveryID);
-    }
 
     /// @notice Closes keep when owner decides that they no longer need it.
     /// Releases bonds to the keep members. Keep can be closed only when
     /// there is no signing in progress or requested signing process has timed out.
     /// @dev The function can be called only by the owner of the keep and only
     /// if the keep has not been already closed.
-    function closeKeep() external onlyOwner onlyWhenActive {
```

Comment:
One last comment: can we reorder functions in `BondedECDSAKeep` into:
- `initialize`
- `closeKeep`
- `terminateKeep`
- `slashForSignatureFraud`

(... the rest stay as-is ...)